src/main/java/org/stellar/sdk/StrKey.java

```diff
@@ -53,16 +51,7 @@ public static String encodeStellarMuxedAccount(MuxedAccount account) {
         if (account.getDiscriminant().equals(CryptoKeyType.KEY_TYPE_ED25519)) {
             return encodeStellarAccountId(account.getEd25519().getUint256());
         } else if (account.getDiscriminant().equals(CryptoKeyType.KEY_TYPE_MUXED_ED25519)) {
-            ByteArrayOutputStream byteStream = new ByteArrayOutputStream();
-
-            try {
-                account.getMed25519().encode(new XdrDataOutputStream(byteStream));
-            } catch (IOException e) {
-                throw new IllegalArgumentException("invalid muxed account", e);
-            }
-
-            char[] encoded = encodeCheck(VersionByte.MUXED_ACCOUNT, byteStream.toByteArray());
-            return String.valueOf(encoded);
+            return encodeStellarAccountId(account.getMed25519().getEd25519().getUint256());
```

Comment:
Instead of modifying `encodeStellarMuxedAccount` to internally transform to an `AccountId` I would add a method to explicitly transform Muxed Accounts into account IDs. This is more explicit and there is no magic involved.